zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java

```diff
@@ -56,6 +57,60 @@
     V convert(JsonParser parser, Supplier<String> contentString) throws IOException;
   }
 
+  public interface RequestSupplier {
+    /**
+     * Creates a new {@link HttpRequest} to send. If this {@link HttpRequest} is not closed, it
```

Comment:
can you add javadoc link on closed? sorry does request have a close function? I don't quite get this part.